### PR TITLE
Added Media Notifications UITests 

### PR DESCRIPTION
**Pull request by sv-ohorvath**
- updated the method to verify system notifications and moved it to a NotificationRobot
- added new tests for media notifications, using that method
- added media pages and event listeners to verify the media playback state

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by AaronMT** [comment]
https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/5194309890971898532


```Kotlin
java.lang.NullPointerException: Attempt to invoke virtual method 'void androidx.test.uiautomator.UiObject2.click()' on a null object reference
at org.mozilla.fenix.ui.robots.BrowserRobot.clickMediaPlayerPauseButton(BrowserRobot.kt:308)
at org.mozilla.fenix.ui.MediaNotificationTest$audioPlaybackSystemNotificationTest$4.invoke(MediaNotificationTest.kt:70)

```

and 
```Kotlin
androidx.test.uiautomator.UiObjectNotFoundException: UiSelector[CONTAINS_DESCRIPTION=Play, RESOURCE_ID=android:id/action0]
at androidx.test.uiautomator.UiObject.click(UiObject.java:416)
at org.mozilla.fenix.ui.robots.NotificationRobot.clickMediaSystemNotificationControlButton(NotificationRobot.kt:44)
at org.mozilla.fenix.ui.MediaNotificationTest$audioPlaybackSystemNotificationTest$3.invoke(MediaNotificationTest.kt:68)
```

**Comment by AaronMT** [comment]
I'll try and get a review in this week. One note, I saw the binary file sizes and we don't want to use such large binaries.

There's smaller binaries that live in mozilla-central we can use (that live in /dom/media) https://hg.mozilla.org/mozilla-central/file/tip/dom/media/test, e.g, ~13kb https://hg.mozilla.org/mozilla-central/raw-file/485e623584d63bbaba3a8b8267bb1d6e55c68042/dom/media/test/owl-short.mp3

and https://hg.mozilla.org/mozilla-central/raw-file/tip/dom/media/test/variable-samplerate.ogg

File: app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import org.junit.Rule
+import org.junit.Test
+import org.mozilla.fenix.helpers.AndroidAssetDispatcher
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.helpers.TestAssetHelper
+import org.mozilla.fenix.ui.robots.navigationToolbar
+
+/**
+ *  Tests for verifying basic functionality of media notifications
+ */
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt]: We should add a bullet point for explaining what each test does in the summary header here.

**Comment by sv-ohorvath** [comment]
> I'll try and get a review in this week. One note, I saw the binary file sizes and we don't want to use such large binaries.
> 
> There's smaller binaries that live in mozilla-central we can use (that live in /dom/media) https://hg.mozilla.org/mozilla-central/file/tip/dom/media/test, e.g, ~13kb https://hg.mozilla.org/mozilla-central/raw-file/485e623584d63bbaba3a8b8267bb1d6e55c68042/dom/media/test/owl-short.mp3
> 
> and https://hg.mozilla.org/mozilla-central/raw-file/tip/dom/media/test/variable-samplerate.ogg

Unfortunately, I didn't find in there anything long enough for the notification to show up (it needs to ~2-3 seconds to appear) and have a decent size. So I googled a bit and found 2 other smaller files, I can try and make them even smaller by cutting them a bit if needed.

**Comment by sv-ohorvath** [comment]
Added some more changes and extended the test suite. I'll run the checks a few times before I'll ask for a new review.

**Comment by sv-ohorvath** [comment]
@AaronMT this is ready for review again when you'll have some spare time. Thank you!

File: app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import org.mozilla.fenix.ui.robots.browserScreen
+import org.mozilla.fenix.ui.robots.homeScreen
+import org.mozilla.fenix.ui.robots.mDevice
+import org.mozilla.fenix.ui.robots.navigationToolbar
+
+/**
+ *  Tests for verifying basic functionality of media notifications:
+ *  - video and audio playback system notifications appear and can pause/play the media content
+ *  - a media notification icon is displayed on the homescreen for the tab playing media content
+ */
```
Comment (rpappalax) [inline app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt]: maybe add a comment, note: this test only verifies media notification icon, not media itself

File: app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(audioTestPage.url) {
+            verifyPageContent(audioTestPage.content)
+            clickMediaPlayerPlayButton()
+            waitForPlaybackToStart()
+        }.openHomeScreen {
+            verifyTabMediaControlButtonState("Pause")
+            clickTabMediaControlButton()
+            verifyTabMediaControlButtonState("Play")
```
Comment (rpappalax) [inline app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt]: per our conversation, proceed to tab and verify if delay still happens w/ debugger. Try mDeviceWaitNotNul, but may need to add an idlingResource here.  @AaronMT can help

File: app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt (hunk 2 above)
Comment (sv-ohorvath) [reply app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt]: done

File: app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt (hunk 3 above)
Comment (sv-ohorvath) [reply app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt]: added the step to verify the media state in the tab.

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt [hunk 4]
```diff
@@ -289,6 +284,30 @@ class BrowserRobot {
...
+                    .clazz("android.widget.Button")
+                    .textContains("Play")
+            ),
+            waitingTime
+        )
+        mediaPlayerPlayButton("Play").click()
+    }
+
+    fun clickMediaPlayerPauseButton() {
+        mDevice.wait(
```
[the remark this answers is not in the document]
Comment (sv-ohorvath) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt]: the method is removed

**Pull request merged**